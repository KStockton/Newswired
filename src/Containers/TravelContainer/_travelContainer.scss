.Travel-header {
  font-size: 1.9rem;
  padding: 36px 0px; 
  box-shadow: 2px 2px 2px #cab6b6;
}

.TravelContainer-wrapper {
  width: 1000px;
  display: flex;
  overflow: scroll;
  width: 100vw;
  }
