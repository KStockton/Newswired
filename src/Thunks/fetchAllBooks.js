import * as action from '../actions/index'



export const fetchAllBooks = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const books = await response.json()
      dispatch(action.isLoading(false));
      dispatch(action.getAllBooks(books))
    } catch (error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}