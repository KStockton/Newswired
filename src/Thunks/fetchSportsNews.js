import * as action from '../actions/index';


export const fetchSportsNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      dispatch(action.getSportsNews(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'sportsNews'))
    }
  }
}