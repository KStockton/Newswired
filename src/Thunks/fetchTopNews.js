import * as action from '../actions/index';


export const fetchTopNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      dispatch(action.getTopNews(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'topNews'))
    }
  }
}

