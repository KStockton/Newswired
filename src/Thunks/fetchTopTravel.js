import * as action from '../actions/index';


export const fetchTopTravel = (url) => {
console.log('url', url)
  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      console.log('data', data)
      dispatch(action.getTopTravel(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}

