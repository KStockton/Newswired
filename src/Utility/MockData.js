export const uncleanBooks = {
  results: {
    books: [{
              title: 'Becoming', 
              description: 'The former first lady describes how she balanced work',
              primary_isbn10: 1234,
              weeks_on_list: 20,
              book_image: 'cool.com',
              rank: 1,
              contributor: 'by Michelle Obama',
              dagger: 0
            }]
  }
}

export const cleanBooks = [{
  id: 1234,
  rank: 1,
  author: 'by Michelle Obama',
  bookImage: 'cool.com',
  description: 'The former first lady describes how she balanced work',
  title: 'Becoming',
  weeksOnList: 20,
  isFavorited: false,
}]

export const mockTopNews = [
  {
    title: 'Syria',
    id: 232
  }, {
    titel: 'Trump loses hair',
    id: 45
  }]

  export const mockNewsError = ''

  export const mockRealNewsError = 'something went wrong'

  export const mockTravel = [{title:'New York'},{title:'Boston'}]

  export const mockRealTravelError = 'No Travel Here'