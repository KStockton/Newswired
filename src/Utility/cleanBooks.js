export const cleanBooks = (uncleanBooks) => {
 const cleanedBooks = uncleanBooks.results.books.map(book => {
    return {
        rank: book.rank,
        author: book.author,
        bookImage: book.book_image,
        description: book.description,
        title: book.title
      }
  })
  return cleanedBooks
}