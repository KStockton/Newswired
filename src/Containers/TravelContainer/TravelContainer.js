import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchTopTravel } from '../../Thunks/fetchTopTravel';
import PropTypes from 'prop-types';
import Card from '../../Components/Card/Card';
import {Link} from 'react-router-dom';
import Loading from '../../Components/Loading/Loading';
const shortid = require('shortid')
const API_KEY =`${process.env.REACT_APP_NEWSAPI_API_KEY}`

export class TravelContainer extends Component {

 componentDidMount() {
    if(this.props.allTravel.length === 0){
      const travelUrl = `https://newsapi.org/v2/everything?q=travel-news&language=en&page=1&domains=vice.com&apiKey=${API_KEY}`
      this.props.fetchTopTravel(travelUrl)
    }
  }
 
  displayTravel = () => {
  
   return this.props.allTravel.map(location => {
    return (<Card {...location} key={shortid.generate()}/>)
   })
  }
  
  render() {
    const {error, allTravel}= this.props
    const errorMsg = 
                    <div>
                      <h2>{error}</h2>
                      <Link to="/Options">Back</Link>
                    </div>

    const displayTravel = this.displayTravel()
   
    return (
      <section>
        <h2 className="Travel-header">Vice Travel News</h2>
        {(error !== '' && allTravel.length === 0) && errorMsg}
        {(error === '' && allTravel.length === 0 ) && <Loading/>}
        <div className="TravelContainer-wrapper Card-wrapper">
          {displayTravel}
        </div>
      </section>
    )
  }
}

TravelContainer.propTypes = {
  allTravel: PropTypes.array,
  error: PropTypes.string,
  fetchTopTravel: PropTypes.func
}

export const mapStateToProps = (state) => ({
  allTravel: state.allTravel,
  error: state.error
})

export const mapDispatchToProps = (dispatch) => ({
  fetchTopTravel: (url) => dispatch(fetchTopTravel(url)),
})

export default connect(mapStateToProps, mapDispatchToProps)(TravelContainer)
