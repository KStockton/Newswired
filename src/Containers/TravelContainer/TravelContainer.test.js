import React from 'react';
import { TravelContainer, mapStateToProps, mapDispatchToProps} from './TravelContainer';
import { shallow } from 'enzyme';
import * as mockData from '../../Utility/MockData';
import { fetchTopTravel } from '../../Thunks/fetchTopTravel';
import { Link } from 'react-router-dom';
jest.mock('../../Thunks/fetchTopTravel');

describe('TravelContainer', () => {
  let wrapper;
  let mockTravel;  
  let mockError; 
  let mockfetchTravel; 

  beforeEach(() => {

    mockTravel = mockData.mockTravel;
    mockfetchTravel = jest.fn();
    mockError = 'something went wrong';

    wrapper = shallow(<TravelContainer 
      allTravel={mockTravel}
      error={mockError}
      fetchTopTravel={mockfetchTravel}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an error if error !== empty string ', () => {
    const error = mockData.mockRealTravelError;
    const allTravel = [];

    wrapper.setProps({error});
    wrapper.setProps({allTravel});
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render loading if allTravel.length is 0 and there is no error', () => {
    const error = '';
    const allTravel = [];

    wrapper.setProps({error});
    wrapper.setProps({allTravel});
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it('should call displaySportsNews when rendered', () => {
    wrapper.instance().displayTravel = jest.fn();
    wrapper.update();
    wrapper.instance().render();
   
    expect(wrapper.instance().displayTravel).toHaveBeenCalled();
  });

  it('should render allTravel when the allTravel.length > 0 is passed', () => {
    let travel = wrapper.instance().displayTravel().length;

    expect(travel).toBe(2);
  });

  it('Link should have a prop of /Options with a link if something goes wrong', () => {
    const allTravel = [];

    wrapper.setProps({error: 'something went wrong'});
    wrapper.setProps({allTravel});
    wrapper.update();

    expect(wrapper.find(Link).props().to).toBe('/Options');
  });

  describe('ComponentDidMount', () => {

    it('should not call fetchTopTravel', () => {

      wrapper.instance().componentDidMount();
      expect(fetchTopTravel).not.toHaveBeenCalled();
    });
  });

  it('should call fetchTopTravel when the componentDidMount and allTravel === 0', () => {
    const mockTravel = [];
    const mockError = '';

    wrapper = shallow(<TravelContainer
      allTravel={mockTravel}
      error={mockError}
      fetchTopTravel={mockfetchTravel}
    />);

    wrapper.instance().componentDidMount();
    expect(mockfetchTravel).toHaveBeenCalled();
  });

});


describe('MSTP', () =>{

  it('should map state from mockstate', () => {
    const mockState = { allTravel :[{title:'Golden'}, {title:'Xavier Basketball'}], error: 'oops'};
    const result = mapStateToProps(mockState);
   
    expect(result).toEqual(mockState);
  });

});

describe('MDTP', () =>{

  const mockDispatch = jest.fn();
  const mappedProps = mapDispatchToProps(mockDispatch);

  it('should dispatch action when correct params passed', () => {
    const mockUrl = 'www.cool.com';
    const actionToDispatch = fetchTopTravel(mockUrl);
   
    mappedProps.fetchTopTravel(mockUrl);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  }) ;
});