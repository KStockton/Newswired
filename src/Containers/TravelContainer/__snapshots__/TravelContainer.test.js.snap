// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelContainer should have an error if error !== empty string  1`] = `
<section>
  <h2
    className="Travel-header"
  >
    Vice Travel News
  </h2>
  <div>
    <h2>
      No Travel Here
    </h2>
    <Link
      to="/Options"
    >
      Back
    </Link>
  </div>
  <div
    className="TravelContainer-wrapper Card-wrapper"
  />
</section>
`;

exports[`TravelContainer should match the snapshot 1`] = `
<section>
  <h2
    className="Travel-header"
  >
    Vice Travel News
  </h2>
  <div
    className="TravelContainer-wrapper Card-wrapper"
  >
    <Card
      key="i7NKjTrh4"
      title="New York"
    />
    <Card
      key="z5BwmowWqC"
      title="Boston"
    />
  </div>
</section>
`;

exports[`TravelContainer should render loading if allTravel.length is 0 and there is no error 1`] = `
<section>
  <h2
    className="Travel-header"
  >
    Vice Travel News
  </h2>
  <Loading />
  <div
    className="TravelContainer-wrapper Card-wrapper"
  />
</section>
`;
